ngraph/src/ngraph/op/swish.cpp

```diff
@@ -0,0 +1,141 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/swish.hpp"
+#include "ngraph/attribute_visitor.hpp"
+#include "ngraph/op/constant.hpp"
+
+#include "ngraph/runtime/host_tensor.hpp"
+#include "ngraph/runtime/reference/swish.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+constexpr NodeTypeInfo op::v4::Swish::type_info;
+
+op::v4::Swish::Swish(const Output<Node>& arg)
+    : Op({arg})
+{
+    constructor_validate_and_infer_types();
+}
+
+op::v4::Swish::Swish(const Output<Node>& arg, const Output<Node>& beta)
+    : Op({arg, beta})
+{
+    constructor_validate_and_infer_types();
+}
+
+bool op::v4::Swish::visit_attributes(AttributeVisitor& visitor)
+{
+    return true;
+}
+
+void op::v4::Swish::validate_and_infer_types()
+{
+    auto inputs_count = input_values().size();
+    NODE_VALIDATION_CHECK(this,
+                          inputs_count == 1 || inputs_count == 2,
+                          "Swish must have 1 or 2 inputs, but it has: ",
+                          inputs_count);
+
+    if (inputs_count == 2)
+    {
+        NODE_VALIDATION_CHECK(this,
+                              input_value(0).get_element_type() ==
+                                  input_value(1).get_element_type(),
+                              "Swish inputs must have the same type but they are: ",
+                              input_value(0).get_element_type(),
+                              " and ",
+                              input_value(1).get_element_type());
+        if (input_value(1).get_partial_shape().is_static())
+        {
+            NODE_VALIDATION_CHECK(this,
+                                  input_value(1).get_shape().size() == 0,
```

Comment:
You check only rank, so I suggest to check that rank is static and then check that `rank.get_length() == 0`

Reply:
Good point. Fixed.